patches/api/0421-Add-source-block-to-the-BlockGrowEvent.patch

```diff
@@ -0,0 +1,100 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Lulu13022002 <41980282+Lulu13022002@users.noreply.github.com>
+Date: Sun, 9 Jul 2023 18:42:33 +0200
+Subject: [PATCH] Add source block to the BlockGrowEvent
+
+
+diff --git a/src/main/java/org/bukkit/event/block/BlockFormEvent.java b/src/main/java/org/bukkit/event/block/BlockFormEvent.java
+index 0106780414abf8cdacd07b8e1435d619bf96bdb3..1bea446e76cb963099263d38f568d80fddf159b3 100644
+--- a/src/main/java/org/bukkit/event/block/BlockFormEvent.java
++++ b/src/main/java/org/bukkit/event/block/BlockFormEvent.java
+@@ -27,9 +27,15 @@ public class BlockFormEvent extends BlockGrowEvent {
+     private static final HandlerList handlers = new HandlerList();
+ 
+     public BlockFormEvent(@NotNull final Block block, @NotNull final BlockState newState) {
+-        super(block, newState);
++        this(block, null, newState); // Paper
+     }
+ 
++    // Paper start
++    public BlockFormEvent(@NotNull final Block block, @org.jetbrains.annotations.Nullable final Block source, @NotNull final BlockState newState) {
++        super(block, source, newState);
++    }
++    // Paper end
++
+     @NotNull
+     @Override
+     public HandlerList getHandlers() {
+diff --git a/src/main/java/org/bukkit/event/block/BlockGrowEvent.java b/src/main/java/org/bukkit/event/block/BlockGrowEvent.java
+index 73d5375d2f67ee00204d80308828cd9f2b2f17e8..13d03f3d59ef147b48a3da3e5287960199915f7a 100644
+--- a/src/main/java/org/bukkit/event/block/BlockGrowEvent.java
++++ b/src/main/java/org/bukkit/event/block/BlockGrowEvent.java
+@@ -24,12 +24,21 @@ import org.jetbrains.annotations.NotNull;
+ public class BlockGrowEvent extends BlockEvent implements Cancellable {
+     private static final HandlerList handlers = new HandlerList();
+     private final BlockState newState;
++    private final Block source; // Paper
+     private boolean cancelled = false;
+ 
++    @Deprecated // Paper
+     public BlockGrowEvent(@NotNull final Block block, @NotNull final BlockState newState) {
++        this(block, null, newState); // Paper
++    }
++
++    // Paper start
++    public BlockGrowEvent(@NotNull final Block block, @org.jetbrains.annotations.Nullable final Block source, @NotNull final BlockState newState) { // Paper
```

Comment:
Add `@Internal` here. I think its good we do that for new event constructors to further emphasize that they can change whenever.